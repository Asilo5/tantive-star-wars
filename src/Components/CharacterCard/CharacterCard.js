import React from 'react';
import './CharacterCard.scss';
import { FaJedi } from "react-icons/fa";

const CharacterCard = ({ id, name, homeworld, population, species, films, toggleFavourites }) => {
  let theFilm = films.map((film, i) => <p key={i}>{film}</p>)
  return (
      <section className='card'>
        <p className='favourite-icon' onClick={() => toggleFavourites(id)}><FaJedi/></p>
        <h2>{name}</h2>
        <h3>Homeworld:</h3>
        <p>{homeworld}</p>
        <h3>Population:</h3>
        <p>{population}</p>
        <h3>Species: </h3>
        <p>{species}</p>
        <h3>Seen in:</h3>
        {theFilm}
      </section>
  ) 
}

export default CharacterCard;