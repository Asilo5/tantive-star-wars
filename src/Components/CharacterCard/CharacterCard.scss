.card {
    width: 37%;
    border: 1px solid green;
    color: green;
    margin: 2em;
    border-radius: 10px;
    .favourite-icon {
        font-size: 40px;
        cursor: pointer;
        &:hover {
            color: gold;
        }
    }
    h2 {
        font-size: 3em;
        margin-top: -0.5em;
    }
    h3 {
        font-size: 27px;
        text-decoration: underline;
    }
    p {
        font-size: 24px;
        margin: 17px;
    }
}