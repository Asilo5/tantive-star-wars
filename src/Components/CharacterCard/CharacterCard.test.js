import React from 'react';
import { shallow } from 'enzyme';
import CharacterCard from './CharacterCard';

describe('CharacterCard', () => {

    const wrapper = shallow(<CharacterCard name = 'Leia'
                                           homeWorld = 'Alderan'
                                           population = '200000000000'
                                           species = 'Human'
                                           films = {[
                                               "The Empire Strikes Back",
                                                "Revenge of the Sith",
                                                "Return of the Jedi"
                                            ]}
                                                 />)

    it('should match snapshot with all information passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
})