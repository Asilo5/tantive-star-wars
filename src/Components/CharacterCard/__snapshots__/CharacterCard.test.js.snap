// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should match snapshot with all information passing in correctly 1`] = `
<section>
  <h2>
    Leia
  </h2>
  <p>
    Homeworld: 
    Alderan
  </p>
  <p>
    Population: 
    200000000000
  </p>
  <p>
    Species: 
    Human
  </p>
  <p>
    Seen in:
    The Empire Strikes Back
    Revenge of the Sith
    Return of the Jedi
  </p>
</section>
`;
